registry/container/trie.go

```diff
@@ -51,8 +51,8 @@ func (r *Trie) MatchUri(uri route.Uri) *route.EndpointPool {
 	var lastPool *route.EndpointPool
 
 	for {
-		pathParts := parts(key)
-		SegmentValue := pathParts[0]
+		before, after, found := strings.Cut(key, "/")
```

Comment:
Do we need the new variables? `before` can simply become `segmentValue` (notice: lower-case :smile:). I'm not so sure for `key` but it looks like we could also assign that one directly (but please double-check that).

Bonus points for not allocating the remaining variables per-run but allocating them before and re-using them in each run.

This applies to all similar for-loops.

Reply:
Fair point. Thanks. I will have another pass.

Reply:
Done. Please have another look.

Reply:
That's the bonus point: allocating in the loop should be avoided entirely if possible.

Reply:
We agreed that this is beyond the scope of this PR.

Reply:
maybe nits, but the `parts` func was probably introduced to avoid code duplication which is now reintroduced.
if you must remove it for performance reasons, I'd opt for making the "/" at least a const like `const pathSeparator = "/"` or something.

Reply:
Agreed to keep it the way it is and push the makeover to a later refactoring.

Reply:
For completeness' sake a quick summary of what we discussed:
1. the `"/"` literal is strewn across the rest of the code. Pulling it out only in the `Trie` would not help significantly.
2. The `parts` function would then look as follows:
```go
func parts(key string) (before, after, found) {
    return strings.Cut(key, "/")
}
```
This does not bring much clarity to what is actually done. Because the whole loop and lookup logic is repeated 5 times, reducing repetition in this one spot will not help much, but adds yet another indirection for the reader of the code.
3. We agree that the `Trie` implementation is 70% repetition of almost the same code and should eventually be refactored. This PR does not aim to do that, just to remove the unnecessary allocation.